docs/develop/topics/i18n.md

```diff
@@ -3,25 +3,54 @@
 ## Babel
 
 ## Marking strings for translation
+In order to translate strings, they have to be marked. This ensures that they will be picked up by the extraction process later on. Only strings which are visible in the UI have to be translated.
+
+### Python
+Marking strings in python is done via the `Flask-BabelEx` package. First, make sure to import the package and the `*gettext` function. We usually import it as `_` to be consistent throughout different packages.
 
 ```python
 from flask_babelex import lazy_gettext as _
-_("Translated string")
 ```
 
-With string interpolation
+Now strings can be marked for translation:
+```python
+error_message = _("Invalid scheme")
+```
 
+Certain data, identifiers or keywords do not have to be translated. This can be achieved with string interpolation:
 ```python
 _("Invalid {scheme}").format(scheme)
 ```
 
-Adding notes to translators
-
+Adding notes for translators helps them to better understand the context where this string will be used:
 ```python
 # NOTE: This is displayed in ....
 _("Invalid {scheme}").format(scheme)
 ```
 
+#### Jinja Templates
+The `gettext` function is made available in jinja templates as well:
+```js
+<p>
+    {{ _('Only published versions are displayed.') }}
+</p>
+```
+
+### JavaScript
+Marking strings in JavaScript is done via the `i18next` module. First, make sure to import `i18next` from the module.
+```js
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+```
+
+Now strings can be marked for translation:
+```js
+<p>
+    {i18next.t('Only published versions are displayed.')}
+</p>
+```
+
```

Comment:
You could add also here: some examples of using a [Trans component](https://react.i18next.com/latest/trans-component) :

Trans component gives you a lot of power by letting you interpolate or translate complex React elements.
e.g.
`import { Trans } from '@translations/i18next';`
```
    const text = (
      <Trans>
        The record can <b>only</b> be accessed by <b>users specified</b> in the
        permissions.
      </Trans>
    );
 ```
 
 OR:
 ```
     const text = (
      <Trans
        defaults="On <bold>{{ date }}</bold> the record and the files will automatically be made publicly accessible. Until then, the record and the files can <bold>only</bold> be accessed by <bold>users specified</bold> in the permissions."
        values={{ date: fmtDate }}
        components={{ bold: <b /> }}
      />
    );

Reply:
in this case, I will just mention this component, with a link to the documentation of i18next, since it is much more explicit on the subject and I would prefer not to duplicate it here

Reply:
Added these examples, one for pluralization (should be the most important ones) and the link to the original docs